domenetjenester/mottak/src/main/java/no/nav/foreldrepenger/mottak/dokumentpersiterer/impl/søknad/v3/SøknadOversetter.java

```diff
@@ -719,11 +728,28 @@ private Arbeidsgiver oversettArbeidsgiver(no.nav.vedtak.felles.xml.soeknad.uttak
         throw new IllegalStateException("Ukjent arbeidsgiver type " + arbeidsgiverFraSøknad.getClass());
     }
 
+    private OppgittOpptjeningBuilder flettOppgittOpptjening(Opptjening opptjening, OppgittOpptjening eksisterendeOppgittOpptjening) {
+        // Bygger ny opptjening fra gammelt grunnlag for å ta vare på gamle opplysninger, så lenge equals metoder er rett vil ikke dette gi dobble innslag
+        var flettetBuilder = OppgittOpptjeningBuilder.oppdater(Optional.of(eksisterendeOppgittOpptjening));
+        var opptjeningFraNySøknad = mapOppgittOpptjening(opptjening).build();
+        opptjeningFraNySøknad.getFrilans().ifPresent(flettetBuilder::leggTilFrilansOpplysninger);
+        opptjeningFraNySøknad.getAnnenAktivitet().stream()
+            .filter(aa -> !eksisterendeOppgittOpptjening.getAnnenAktivitet().contains(aa))
+            .forEach(flettetBuilder::leggTilAnnenAktivitet);
+        opptjeningFraNySøknad.getEgenNæring()
+            .stream().filter(en -> !eksisterendeOppgittOpptjening.getEgenNæring().contains(en))
+            .forEach(flettetBuilder::leggTilEgenNæring);
```

Comment:
Her vil vi kanskje heller fjerne det som finnes fra før? Ditto for annen aktivitet.
Jeg ser at equals i OppgittEgenNæring vil si at den ikke finnes dersom noe er ulikt. Men vil kanskje heller ha det slik at hvis orgnummer finnes fra før så slett gammel og legg til ny.